python/lsst/ip/isr/calibType.py

```diff
@@ -0,0 +1,630 @@
+    def readText(cls, filename):
+        """Read calibration representation from a yaml/ecsv file.
+
+        Parameters
+        ----------
+        filename : `str`
+            Name of the file containing the calibration definition.
+
+        Returns
+        -------
+        calib : `~lsst.ip.isr.IsrCalibType`
+            Calibration class.
+        """
+        data = ''
+        try:
+            data = Table.read(filename, format='ascii.ecsv')
+        except Exception as e1:
+            try:
+                with open(filename, 'r') as f:
+                    data = yaml.load(f, Loader=yaml.CLoader)
+            except Exception as e2:
+                raise e2(f"Failed to read as ecsv as well: {e1}")
+
+        if isinstance(data, dict):
+            return cls().fromDict(data)
+        else:
+            schema = afwTable.Schema()
+            keyDict = dict()
+            for colName in data.columns:
+                try:
+                    keyDict[colName] = schema.addField(colName,
+                                                       type=data[colName].dtype.type)
+                except KeyError:
+                    keyDict[colName] = schema.addField(colName, type="String", size=32)
+
+            table = afwTable.BaseCatalog(schema)
+            table.resize(len(data))
+
+            for colName in data.columns:
+                key = keyDict[colName]
+                if key.getTypeString() == 'String':
+                    for rowNum, row in enumerate(data[colName]):
+                        table[rowNum][key] = row
+                else:
+                    table[key] = data[colName]
+
+            metadata = PropertyList()
+            for k, v in data.meta.items():
+                metadata[k] = v
+            table.setMetadata(metadata)
+
+            return cls().fromTable([table])
+
+    def writeText(self, filename):
+        """Write the calibration data to a text file.
+
+        Parameters
+        ----------
+        filename : `str`
+            Name of the file to write.
+
+        Returns
+        -------
+        used : `str`
+            The name of the file used to write the data.
+
+        Raises
+        ------
+        RuntimeError :
+            Raised if filename does not end in a known extension
+
+        Notes
+        -----
+        The file is written to YAML/ECSV format and will include any
+        associated metadata.
+        """
+        outDict = self.toDict()
+        if filename.lower().endswith((".yaml")):
+            with open(filename, 'w') as f:
+                yaml.dump(outDict, f)
+        elif filename.lower().endswith((".ecsv")):
+            tableList = self.toTable()
+            tableX = tableList[0]
+
+            table = tableX.asAstropy()
+            table.meta = tableX.getMetadata().toDict()
+
+            # ECSV doesn't support multiple tables per file, so we can only
+            # write the first table.
+            table.write(filename, format="ascii.ecsv")
```

Comment:
I was thinking about Linearizer. It might be that we could sneak the look up table into the yaml metadata and use the table for the main component used by all.

Shouldn't we warn if we have multiple tables and can't write one out?

Reply:
I've changed this to raise if there is more than a single table, because not writing something you had is certainly an error.
I think we have to support multi-table things for Linearizer (Brighter-Fatter I think also is a place where this may happen).  I would prefer not to hide it in the metadata, but I don't have a strong feeling.